### ISSUE TITLE
Wrong text positioning?

### ISSUE THREAD

**Issue by sdelaysam** [oxygine/oxygine-framework#113]
Hi, @frankinshtein 

Working with TTF fonts in Oxygine.
Looks like text positioning is wrong, but text rect calculation is correct.
I'd say that font's descent is ignored.
Please look at screenshot.

<img width="755" alt="screen shot 2017-12-04 at 21 55 01" src="https://user-images.githubusercontent.com/168758/33559169-8231f47e-d93e-11e7-9e4a-88102fb15d2b.png">

Code for drawing text and red rectangle:
```cpp

_textfield->setSize(getSize()); // <-- size of white background

const Rect& rect = _textfield->getTextRect(); // <-- rect of red overlay
_textrect->setPosition(rect.pos);
_textrect->setSize(rect.size);
```

**Comment by frankinshtein** [comment]
0. it is not related with TTF
1. check 
oxygine examples/Demo/Text 
it has all possible combinations demo
2. Do you know about TextField::setVerticalAlign?

**Comment by sdelaysam** [comment]
0. I'm afraid it is related with how you work with metrics provided by freetype library
1. i did it
2. yes, sure. i'm using `VALIGN_MIDDLE`. Otherwise red rect won't be vertically aligned.

**Comment by sdelaysam** [comment]
1. White rect - size of TextField. 
2. Rect red - bounds of text itself. 
Problem: Glyphs are out of text rect.

**Comment by frankinshtein** [comment]
At this moment I don't know what metrick should be used from TTF. Do you know?

in your case could help for example
TextField::setBaselineScale (0.65); it will align text like in your screenshot

**Comment by sdelaysam** [comment]
Yep, with `TextField::setBaselineScale (0.65);` works fine!
Not sure about glyph metrics, will let you know if find something.

**Comment by sdelaysam** [comment]
Obviously, there is a more elegant way to achieve this, but the following worked for me.
In `ResFontFT.cpp` from Line 170:
```cpp
FT_GlyphSlot  slot = face->glyph;
short yCorrection = static_cast<short>((face->descender - (face->bbox.yMax - face->bbox.yMin - face->height) / 2) >> 6);
...
g.offset_x = slot->bitmap_left;
g.offset_y = -slot->bitmap_top + yCorrection;
...
```

**Comment by sdelaysam** [comment]
#115 works fine for me
(together with https://github.com/oxygine/oxygine-freetype/pull/5)
